test/commands/auth/list.test.ts

```diff
@@ -5,65 +5,45 @@
   const testData = new MockTestOrgData();
-  let authInfoStub: StubbedType<AuthInfo>;
+  testData.aliases = ['TestAlias'];
 
   async function prepareStubs(forceFailure = false) {
-    const authFields = await testData.getConfig();
-    authInfoStub = stubInterface<AuthInfo>($$.SANDBOX, {
-      getFields: () => authFields,
-      getConnectionOptions: () => ({ accessToken: authFields.accessToken }),
-      isJwt: () => true,
-      isOauth: () => false,
-    });
-
-    let auth = {
-      username: authFields.username,
-      aliases: ['TestAlias'],
-      orgId: testData.orgId,
-      instanceUrl: testData.instanceUrl,
-      oauthMethod: forceFailure ? 'unknown' : 'jwt',
-    } as OrgAuthorization;
-
+    await $$.stubAuths(testData);
+    $$.stubAliases({ TestAlias: testData.username });
     if (forceFailure) {
-      auth = { ...auth, ...{ error: 'decrypt error' } };
-    } else {
-      stubMethod($$.SANDBOX, AuthInfo, 'create').callsFake(async () => authInfoStub);
+      $$.SANDBOX.stub(AuthInfo, 'create').throws(new Error('decrypt error'));
     }
-    $$.SANDBOX.stub(AuthInfo, 'listAllAuthorizations').resolves([auth]);
   }
 
-  test
-    .do(async () => prepareStubs())
-    .stdout()
-    .command(['auth:list', '--json'])
-    .it('should show auth files', (ctx) => {
-      const auths = parseJson<Array<OrgAuthorization & { alias: string }>>(ctx.stdout).result;
-      expect(auths[0].alias).to.deep.equal('TestAlias');
-      expect(auths[0].username).to.equal(testData.username);
-      expect(auths[0].instanceUrl).to.equal(testData.instanceUrl);
-      expect(auths[0].orgId).to.equal(testData.orgId);
-      expect(auths[0].oauthMethod).to.equal('jwt');
-    });
+  it('should show auth files', async () => {
+    await prepareStubs();
+    const list = new List(['--json'], {} as Config);
+    const auths = await list.run();
+    expect(auths[0].alias).to.deep.equal(testData.aliases?.join(',') ?? '');
+    expect(auths[0].username).to.equal(testData.username);
+    expect(auths[0].instanceUrl).to.equal(testData.instanceUrl);
+    expect(auths[0].orgId).to.equal(testData.orgId);
+    expect(auths[0].oauthMethod).to.equal('web');
+  });
 
-  test
-    .do(async () => prepareStubs(true))
-    .stdout()
-    .command(['auth:list', '--json'])
-    .it('should show files with auth errors', (ctx) => {
-      const auths = parseJson<Array<OrgAuthorization & { alias: string }>>(ctx.stdout).result;
-      expect(auths[0].alias).to.deep.equal('TestAlias');
-      expect(auths[0].username).to.equal(testData.username);
-      expect(auths[0].instanceUrl).to.equal(testData.instanceUrl);
-      expect(auths[0].orgId).to.equal(testData.orgId);
-      expect(auths[0].oauthMethod).to.equal('unknown');
-      expect(auths[0].error).to.equal('decrypt error');
-    });
+  it('should show files with auth errors', async () => {
+    await prepareStubs(true);
+    const list = new List(['--json'], {} as Config);
+    const auths = await list.run();
```

Comment:
```suggestion
    const [auth] = await list.run();
```

then you don't have to auths[0] on all the remaining lines

Reply:
Nice - thanks